### PR TITLE
[Wallet] Replace QR scanner screen instead of keeping it in stack

### LINKED ISSUE #3425: [Wallet] iOS redirect on transaction is buggy
### Expected Behavior

User stays on `Review payment` after they entered amount and pressed "Send"

### Current Behavior

User sometimes navigated back to `Send or Request` screen. 
Video: https://celo-org.slack.com/archives/CE48N92M7/p1583445768004700

--

As a part of #3068

### PR DESCRIPTION
**Pull request by i1skn**
## Description

This is needed to unmount the QR code scanner screen, so it does not continue to scan even though a user has moved to other screen.

## Other changes

* Remove `.flowconfig` cause we are using `TypeScript`
* Ensure that `navigator` exists in `navigateBack()`

## Tested

Android device & iPhone

## Related issues

- Fixes #3425

## Backwards compatibility

Yes

## Commit Message
[Wallet] Replace QR scanner screen instead of keeping it in stack

This is needed to unmount the QR code scanner screen, so it does not continue to scan even though a user has moved to other screen.

### CODE REVIEW

**Review by annakaz — APPROVED** [review]
👍 LGTM

File: packages/mobile/src/qrcode/QRScanner.tsx [hunk 1]
```diff
@@ -23,50 +24,31 @@ interface DispatchProps {
...
-  state = {
-    isScanningEnabled: true,
-  }
+  state = {}
 
   camera: RNCamera | null = null
 
   // This method would be called multiple times with the same
   // QR code, so we need to catch only the first one
   onBardCodeDetected = (rawData: any) => {
```
Comment (annakaz) [inline packages/mobile/src/qrcode/QRScanner.tsx]: Nit: not a new typo but "onBarcodeDetected" (removing d)

**Review by jmrossy — APPROVED** [review]
lgtm 👍

File: packages/mobile/src/qrcode/QRScanner.tsx (hunk 1 above)
Comment (i1skn) [reply packages/mobile/src/qrcode/QRScanner.tsx]: Fixed!

**Pull request merged**